internal/corazawaf/transaction.go

```diff
@@ -1403,26 +1405,29 @@ func (tx *Transaction) AuditLog() *auditlog.Log {
 			}
 		case types.AuditLogPartRulesMatched:
 			for _, mr := range tx.matchedRules {
-				r := mr.Rule()
-				for _, matchData := range mr.MatchedDatas() {
-					al.Messages_ = append(al.Messages_, auditlog.Message{
-						Actionset_: strings.Join(tx.WAF.ComponentNames, " "),
-						Message_:   matchData.Message(),
-						Data_: &auditlog.MessageData{
-							File_:     mr.Rule().File(),
-							Line_:     mr.Rule().Line(),
-							ID_:       r.ID(),
-							Rev_:      r.Revision(),
-							Msg_:      matchData.Message(),
-							Data_:     matchData.Data(),
-							Severity_: r.Severity(),
-							Ver_:      r.Version(),
-							Maturity_: r.Maturity(),
-							Accuracy_: r.Accuracy(),
-							Tags_:     r.Tags(),
-							Raw_:      r.Raw(),
-						},
-					})
+				// Log action is required to log a matched rule on both error log and audit log
+				if mr.Log() {
```

Comment:
one easy way to avoid the breaking change is to not to add `Log` method to the interface and do the assertion here with a `RuleLogger` interface:

```go
type RuleLogger interface {
    Log() bool
}
```

with a comment so breaking change can be added in v4.

Reply:
Thanks! I tried to describe it also close to the commented-out Log() method, to hopefully still easily provide this filtering boolean to the users